addons/account/tests/test_chart_template.py

```diff
@@ -462,3 +462,38 @@ def get_domain(model):
         for model in TEMPLATE_MODELS + sub_models:
             data_after = self.env[model].search(get_domain(model))
             self.assertEqual(data_before[model], data_after)
+
+    def test_unknown_company_fields_skip(self):
+        """ Tests that if a key is not known in the company template data, that key is skipped
+        and no error is raised. """
+
+        def local_get_data(self, template_code):
+            company_3 = self.env['res.company'].search([('name', '=', 'TestCompany3')], limit=1)
+            data = test_get_data(self, template_code)
+            data['res.company'][company_3.id]['unknown_company_key'] = 'unknown_company_value'
+            return data
+
+        company_3 = self.env['res.company'].create({
+            'name': 'TestCompany3',
+            'country_id': self.env.ref('base.be').id,
+        })
+        with patch.object(AccountChartTemplate, '_get_chart_template_data', side_effect=local_get_data, autospec=True):
+            self.env['account.chart.template'].try_loading('test', company=company_3, install_demo=False)
+
+    def test_unknown_company_fields_error(self):
+        """ Tests that if a key is not known in the company template data when the
+        context value 'l10n_check_fields_complete' is set, an error is raised. """
+
+        def local_get_data(self, template_code):
+            company_4 = self.env['res.company'].search([('name', '=', 'TestCompany4')], limit=1)
+            data = test_get_data(self, template_code)
+            data['res.company'][company_4.id]['unknown_company_key'] = 'unknown_company_value'
+            return data
+
+        company_4 = self.env['res.company'].create({
+            'name': 'TestCompany4',
+            'country_id': self.env.ref('base.be').id,
+        })
+        with patch.object(AccountChartTemplate, '_get_chart_template_data', side_effect=local_get_data, autospec=True):
+            with self.assertRaises(ValueError), self.env.cr.savepoint():
+                self.env['account.chart.template'].with_context(l10n_check_fields_complete=True).try_loading('test', company=company_4, install_demo=False)
```

Comment:
A few things:
* no need to do the same setup twice, it's actually better to keep the same to be sure that we are testing the same thing in the two tests
* no need to create a new company
* you can use https://docs.python.org/3/whatsnew/3.10.html#parenthesized-context-managers instead of stacking the context managers (with 2 different syntaxes in one?)
* check the content of the error with `assertRaisesRegex` to be sure of not catching something unexpected

```suggestion
        def local_get_data(self, template_code):
            data = test_get_data(self, template_code)
            data['res.company'][company.id]['unknown_company_key'] = 'unknown_company_value'
            return data

        company = self.company_1
        with patch.object(AccountChartTemplate, '_get_chart_template_data', side_effect=local_get_data, autospec=True):
            # hard fail the loading if the context key is set to ensure `test_all_l10n` works as expected
            with (
                self.assertRaisesRegex(ValueError, 'unknown_company_key'),
                self.env.cr.savepoint(),
            ):
                self.env['account.chart.template'].with_context(l10n_check_fields_complete=True).try_loading('test', company=company, install_demo=False)

            # silently ignore if the field doesn't exist (yet)
            self.env['account.chart.template'].try_loading('test', company=company, install_demo=False)

```